pkgs/stdenv/generic/default.nix

```diff
@@ -223,7 +223,7 @@ let
         # include it in the result, it *is* available to nix-env for queries.
         meta = { }
             # If the packager hasn't specified `outputsToInstall`, choose a default,
-            # namely `p.bin or p.out or p`;
+            # namely `lib.getBin p`;
```

Comment:
The code
```nix
lib.findFirst hasOutput null (["bin" "out"] ++ outs)
```
first tries "bin" output then "out" and then the first output (`outs` at this point is the list of output names).